python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets.py

```diff
@@ -208,3 +209,96 @@ def test_coerced_asset_keys():
     @asset(ins={"input1": AssetIn(asset_key=["Asset", "1"])})
     def asset1(input1):
         assert input1
+
+
+def test_asset_with_io_manager_def():
+    io_manager_inst = InMemoryIOManager()
+
+    @io_manager
+    def the_io_manager():
+        return io_manager_inst
+
+    @asset(io_manager_def=the_io_manager)
+    def the_asset():
+        return 5
+
+    AssetGroup([the_asset]).materialize()
+
+    assert list(io_manager_inst.values.values())[0] == 5
+
+
+def test_multiple_assets_io_manager_defs():
+    io_manager_inst = InMemoryIOManager()
+    num_times = [0]
```

Comment:
lolol python scoping

Reply:
i have done this more times than I care to admit